libraries/AP_HAL_ChibiOS/hwdef/common/hrt.c

```diff
@@ -20,50 +20,49 @@
 #include "hrt.h"
 #include <stdint.h>
 
-static uint64_t timer_base_us64;
-#if CH_CFG_ST_RESOLUTION == 16
-static uint32_t timer_base_us32;
-#endif
-static uint32_t timer_base_ms;
-static volatile systime_t last_systime;
-
 #if CH_CFG_ST_RESOLUTION == 16
-static uint32_t get_systime_us32(void)
+static uint32_t system_time_u32_us(void)
 {
     systime_t now = chVTGetSystemTimeX();
-#if CH_CFG_ST_FREQUENCY != 1000000
+#if CH_CFG_ST_FREQUENCY != 1000000U
     now *= (1000000UL / CH_CFG_ST_FREQUENCY);
 #endif
-    if (now < last_systime) {
-        uint32_t last_u32 = timer_base_us32;
-        timer_base_us32 += (uint32_t)TIME_MAX_SYSTIME;
-        if (timer_base_us32 < last_u32) {
-            timer_base_us64 += ((uint32_t)-1);
-            timer_base_ms += ((uint32_t)-1)/1000;
-        }
-    }
+    static systime_t last_systime;
+    static uint32_t timer_base_us32;
+    uint16_t dt = now - last_systime;
     last_systime = now;
-    return timer_base_us32 + (uint32_t)now;
+    timer_base_us32 += dt;
+    return timer_base_us32;
 }
-
 #elif CH_CFG_ST_RESOLUTION == 32
-static uint32_t get_systime_us32(void)
+static uint32_t system_time_u32_us(void)
 {
-    systime_t now = chVTGetSystemTimeX();
-#if CH_CFG_ST_FREQUENCY != 1000000
+    uint32_t now = chVTGetSystemTimeX();
```

Comment:
Why's this changed from `systime_t`?  The static assert down below seems to guarantee this was OK.

Reply:
fixed